### PR TITLE
Feat/pagination

### PR DESCRIPTION
**Pull request by GeorgeGoodall**
Ticket: https://trello.com/c/vKqJrQ60/1409-pagination-for-large-files?filter=member:georgegoodall8

This ticket add pagination to the errors and no errors pages
It also adds some additional checks on a provided url. by making a head request to the url and checking it looks as it should

### CODE REVIEW

File: src/models/requestData.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
 import getVerboseColumns from '../utils/getVerboseColumns.js'
 import logger from '../utils/logger.js'
+import axios from 'axios'
+import config from '../../config/index.js'
 
 export default class RequestData {
```
Comment (cpcundill) [inline src/models/requestData.js:6]: Maybe this is now inappropriately named? Seems like it wraps the HTTP response for an async request and facilitates the fetching of the response details.  Might be better to split out the request model and response details models?

File: test/unit/requestData.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+      input: {
+        count: 10,
+        current: 5
+      },
+      expected: [1, '...', 4, 5, 6, '...', 10]
+    }
+  ]
+
+  testCases.forEach((testCase, index) => {
```
Comment (cpcundill) [inline test/unit/requestData.test.js:47]: Nice use case for a parameterised test 👏

File: test/unit/requestData.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { pagination } from '../../src/models/requestData'
+import { describe, it, expect } from 'vitest'
+
+// // Tech Debt: we should write some tests around the requestData.js file
```
Comment (cpcundill) [inline test/unit/requestData.test.js:4]: Worth capturing a Trello card for this tech debt and including the link in the comments

**Review by cpcundill — CHANGES_REQUESTED** [review]
A few things to discuss and consider changing before we merge @GeorgeGoodall

File: src/models/requestData.js (hunk 1 above)
Comment (GeorgeGoodall) [reply src/models/requestData.js:6]: we can pull this out into another ticket for now

**Review by cpcundill — APPROVED** [review]

**Pull request merged**